state/protocol/epoch.go

```diff
@@ -10,33 +10,39 @@ type EpochQuery interface {
 
 	// Current returns the current epoch as of this snapshot. All valid snapshots
 	// have a current epoch.
-	Current() CommittedEpoch
+	// Error returns:
+	//   - state.ErrUnknownSnapshotReference - if the epoch is queried from an unresolvable snapshot.
+	//   - generic error in case of unexpected critical internal corruption or bugs
+	Current() (CommittedEpoch, error)
 
 	// NextUnsafe should only be used by components that actively advance the
 	// epoch from flow.EpochPhaseSetup to flow.EpochPhaseCommitted.
 	// NextUnsafe returns the next epoch as of this snapshot. Valid snapshots must
 	// have a next epoch available after the transition to epoch setup phase.
 	//
-	// Returns invalid.Epoch with ErrNextEpochNotSetup in the case that this method
-	// is queried w.r.t. a snapshot within the flow.EpochPhaseStaking phase, or
-	// ErrNextEpochAlreadyCommitted during the flow.EpochPhaseCommitted phase.
-	NextUnsafe() TentativeEpoch
+	// Error returns:
+	//   - ErrNextEpochNotSetup in the case that this method is queried w.r.t. a snapshot within the flow.EpochPhaseStaking phase
+	//   - ErrNextEpochAlreadyCommitted during the flow.EpochPhaseCommitted phase
+	NextUnsafe() (TentativeEpoch, error)
 
 	// NextCommitted returns the next epoch as of this snapshot, only if it has
 	// been committed already (after flow.EpochPhaseCommitted)
 	//
-	// Returns invalid.Epoch with ErrNextEpochNotCommitted in the case that
-	// the current phase is flow.EpochPhaseStaking or flow.EpochPhaseSetup.
-	NextCommitted() CommittedEpoch
+	// Error returns:
+	//   - ErrNextEpochNotCommitted in the case that the current phase is flow.EpochPhaseStaking or flow.EpochPhaseSetup.
```

Comment:
```suggestion
	//   - ErrNextEpochNotCommitted in the case that the current phase is flow.EpochPhaseStaking or flow.EpochPhaseSetup.
	//   - state.ErrUnknownSnapshotReference - if the epoch is queried from an unresolvable snapshot.
	//   - generic error in case of unexpected critical internal corruption or bugs
```